elastic/kibana
# Additional context for log threshold rule

## Summary

Resolves https://github.com/elastic/kibana/issues/146348

- Adds new context variables for count and ratio alerts in Log threshold rule
- Indexes new context variables to AAD
- Adds new context variables to recovered count and ratio alerts

The context variables are added based on group by and filter used when creating the rule.

When the prefix of `group by` or filter (`WITH` condition) includes one of the following, contextual attributes for that entity are added.

- `cloud`
- `host`
- `orchestrator`
- `container`
- `labels`
- `tags`

Following fields are excluded from the context: `*.cpu`, `*.network`, `*.disk`, `*.memory`

In case of ratio alerts without `group by`, the prefix from numerator query is considered to add contextual attributes. In both count and ratio alerts, only positive criteria is used from filter (with `is` or `matches` condition) for adding contextual attributes.

### Manual Testing
1. Create different Log threshold rules (Count, Ratio, With group by, Without group by)
2. Configure action template with `{{context}}`
3. Observe additional context to be included in alert notification
4. Let alerts be recovered
5. Observe additional context to be included in recovery notification
6. Observe additional context fields are indexed in AAD

### Checklist
